Mage.Sets/src/mage/cards/d/DiamondCity.java

```diff
@@ -0,0 +1,118 @@
+package mage.cards.d;
+
+import mage.abilities.Ability;
+import mage.abilities.common.EntersBattlefieldAbility;
+import mage.abilities.condition.Condition;
+import mage.abilities.costs.common.TapSourceCost;
+import mage.abilities.decorator.ConditionalActivatedAbility;
+import mage.abilities.effects.common.counter.AddCountersSourceEffect;
+import mage.abilities.effects.common.counter.MoveCountersFromSourceToTargetEffect;
+import mage.abilities.hint.ConditionHint;
+import mage.abilities.hint.Hint;
+import mage.abilities.mana.ColorlessManaAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.constants.WatcherScope;
+import mage.constants.Zone;
+import mage.counters.CounterType;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.game.permanent.Permanent;
+import mage.target.common.TargetCreaturePermanent;
+import mage.util.CardUtil;
+import mage.watchers.Watcher;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+/**
+ *
+ * @author notgreat
+ */
+public final class DiamondCity extends CardImpl {
+
+    public DiamondCity(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.LAND}, "");
+
+        // Dapper Shieldmate enters the battlefield with a shield counter on it.
+        this.addAbility(new EntersBattlefieldAbility(
+                new AddCountersSourceEffect(CounterType.SHIELD.createInstance(1)),
+                "with a shield counter on it. <i>(If it would be dealt damage " +
+                        "or destroyed, remove a shield counter from it instead.)</i>"
+        ));
+
+        // {T}: Add {C}.
+        this.addAbility(new ColorlessManaAbility());
+
+        // {T}: Move a shield counter from Diamond City onto target creature. Activate only if two or more creatures entered the battlefield under your control this turn.
+        Ability ability = new ConditionalActivatedAbility(Zone.BATTLEFIELD, new MoveCountersFromSourceToTargetEffect(CounterType.SHIELD),
+                new TapSourceCost(), DiamondCityCondition.instance);
+        ability.addTarget(new TargetCreaturePermanent());
+        ability.addHint(DiamondCityCondition.getHint());
+        this.addAbility(ability, new DiamondCityWatcher());
+    }
+
+    private DiamondCity(final DiamondCity card) {
+        super(card);
+    }
+
+    @Override
+    public DiamondCity copy() {
+        return new DiamondCity(this);
+    }
+}
+
+//Based on Celebration Condition/Watcher
+class DiamondCityWatcher extends Watcher {
```

Comment:
Should I also get rid of CelebrationWatcher? It seems like that one's redundant with the standard watcher.

Reply:
Yes, feel free to do that.

Reply:
Hold on, I don't think that works correctly. It needs to evaluate if the permanent was a creature at the time of entering the battlefield, which requires using `getCardType(game)` instead of the static-only `getCardType()`. However, the game object must be the one from the exact time of entering since the game object can change. If the permanent entered as a creature and then becomes a noncreature, it still counts, and there's no way to do that with the `PermanentsEnteredBattlefieldWatcher`.

Reply:
It contains perm copy, so if you use isCreature without game param then it can help?

![IMG_0544](https://github.com/magefree/mage/assets/8344157/c0e9fe82-2b39-4ae4-8da1-a63189fcfd3b)

Reply:
That won't help if the reason it's entering as a creature is something like [[Dance of the Manse]] or [[Opalescence]], right?

Reply:
If it not works then it's a problem of Dance of the Manse and same effects implementation, cause:

> If X is 6 or more, the artifacts and/or enchantments enter the battlefield as 4/4 creatures. They don't enter as noncreature permanents and then become creatures. (2019-10-04)

![shot_240620_105339](https://github.com/magefree/mage/assets/8344157/8fd4cbf4-dab3-4146-96f0-2d3e66e5f766)

Reply:
It's a continuous effect that modifies the permanent entering, and thus that change is not applied directly to the Permanent. That's the whole point of the separation of `getCardType(game)` from `getCardType()`, the latter ignores continuous effects.

Reply:
Well, use PermanentsEnteredBattlefieldWatcher and check perms with game param. There are many usages with same use cases:
![shot_240620_110334](https://github.com/magefree/mage/assets/8344157/b31c7686-4b80-4636-8377-3dcc536589cf)

If you catch bugs with Dance of the Manse or Opalescence then report as new issue -- that's bug will affect all cards and not related to current PR.

Reply:
> Well, use PermanentsEnteredBattlefieldWatcher and check perms with game param.
> If you catch bugs with Dance of the Manse or Opalescence then report as new issue -- that's bug will affect all cards and not related to current PR.

I think that the bug is with `PermanentsEnteredBattlefieldWatcher`'s design. Here's an example where using PermanentsEnteredBattlefieldWatcher fails, but the current code passes:
```
    @Test
    public void testContinuousModifier2() {
        addCard(Zone.BATTLEFIELD, playerA, "Ashaya, Soul of the Wild");
        addCard(Zone.HAND, playerA, "Armory Mice");
        addCard(Zone.HAND, playerA, "Swords to Plowshares");
        addCard(Zone.HAND, playerA, "Black Lotus");

        setStrictChooseMode(true);

        castSpell(1, PhaseStep.PRECOMBAT_MAIN, playerA, "Black Lotus", true);
        setChoice(playerA, "White");
        castSpell(1, PhaseStep.PRECOMBAT_MAIN, playerA, "Armory Mice", true);
        castSpell(1, PhaseStep.PRECOMBAT_MAIN, playerA, "Swords to Plowshares", "Ashaya, Soul of the Wild", true);


        setStopAt(1, PhaseStep.END_TURN);
        execute();
        assertPowerToughness(playerA, "Armory Mice", 3, 1);
        assertGraveyardCount(playerA, 2);
    }
```

Reply:
P.S. 

> That's the whole point of the separation of getCardType(game) from getCardType()

After some research. Outdated logic: permanents persist all actual characteristics (original card + dynamic added abilities to the card) -- e.g. it's return same result for game and non-game methods -- watcher with copies works with it. But it was removed after #7899. So current logic: permanent persist only printed/blueprint characteristics  and separate all dynamic changes data (same as cards does) -- watcher with copies will fail and do not save full permanent state in the time.

![shot_240620_110923](https://github.com/magefree/mage/assets/8344157/126c0dd6-5039-4196-8e87-68887cb609d2)

Reply:
It's funny -- all LKI related code will fails with same use cases -- so LKI's permanents will contains wrong info about dynamic added characteristics. Example: possible bugs with Changeling.

![shot_240620_112954](https://github.com/magefree/mage/assets/8344157/f261a746-badb-4bcf-8735-8d3d89b42b0f)

Reply:
Related issue #12506